### ISSUE TITLE
custom performance metric 

### ISSUE THREAD

**Issue by igoralesey** [topepo/caret#833]
I have a question about the consequences of using a custom performance metric (via "metric" in train and "summaryFunction" in trainControl). 

I can see that the custom metric has an effect on the performance metric for parameter tuning. however, when caret is "Fitting final model on full training set" I see that the custom metric is ignored (i.e. I get the same results for the final model (and fitted/predicted values) as if I had used the standard (RMSE) metric. 

Am I right/wrong? Is that intended? Is there a way of forcing the custom metric to be used in fitting the final model on the full training set? 

An R-script with an example can be found here: [https://www.dropbox.com/s/yuqjd24v0x39hpe/caret-xgboost-custom-metric-icecream-data.R?dl=0](https://www.dropbox.com/s/yuqjd24v0x39hpe/caret-xgboost-custom-metric-icecream-data.R?dl=0) 
At the very end I compare metrics of four different versions; (1) xgboost-standard, (2) xgboost-custom, (3) caret-xgboost-standard, (4) caret-xgboost-custom; all four versions are based on models fitted to the entire training set. I expected (2)=(4), however, (1)=(3)=(4). 

Thanks! 

Igor

**Comment by topepo** [comment]
The custom summary function in `train` is only computed _after_ the model is fit; it is not the loss function that the model is optimizing. `xgboost` can do that (obviously) but I would not expect the results to be the same when one is optimized differently than the other.

**Comment by igoralesey** [comment]
Thanks! I understand. Is there a way within caret to define a custom loss function to be used in caret's model fitting/optimiziation?

**Comment by topepo** [comment]
No. `caret` relies on the underlying model function. If that has an argument for the loss function to optimize, you can pass it in through `train` to the model function. Otherwise, no.

**Comment by igoralesey** [comment]
Thanks a lot!
